lesson6/README.md

```diff
@@ -0,0 +1,147 @@
+
+The `RequestQueue` must be instantiated in your Application as opposed to your Activity. This is so the queue does not get recreated when the activity is (such as when the device is rotated).
+
+Here's the code for a singleton class:
+
+```
+public class MySingleton {
+    private static MySingleton mInstance;
+    private RequestQueue mRequestQueue;
+    private ImageLoader mImageLoader;
+    private static Context mCtx;
+
+    private MySingleton(Context context) {
+        mCtx = context;
+        mRequestQueue = getRequestQueue();
+
+        mImageLoader = new ImageLoader(mRequestQueue,
+                new ImageLoader.ImageCache() {
+            private final LruCache<String, Bitmap>
+                    cache = new LruCache<String, Bitmap>(20);
+
+            @Override
+            public Bitmap getBitmap(String url) {
+                return cache.get(url);
+            }
+
+            @Override
+            public void putBitmap(String url, Bitmap bitmap) {
+                cache.put(url, bitmap);
+            }
+        });
+    }
+
+    public static synchronized MySingleton getInstance(Context context) {
+        if (mInstance == null) {
+            mInstance = new MySingleton(context);
+        }
+        return mInstance;
+    }
+
+    public RequestQueue getRequestQueue() {
+        if (mRequestQueue == null) {
+            // getApplicationContext() is key, it keeps you from leaking the
+            // Activity or BroadcastReceiver if someone passes one in.
+            mRequestQueue = Volley.newRequestQueue(mCtx.getApplicationContext());
+        }
+        return mRequestQueue;
+    }
+
+    public <T> void addToRequestQueue(Request<T> req) {
+        getRequestQueue().add(req);
+    }
+
+    public ImageLoader getImageLoader() {
+        return mImageLoader;
+    }
+}
+```
+
+Using your singleton to access your application-wide queue:
+
+```
+// Get a RequestQueue
+RequestQueue queue = MySingleton.getInstance(this.getApplicationContext()).
+    getRequestQueue();
+
+// ...
+
+// Add a request (in this example, called stringRequest) to your RequestQueue.
+MySingleton.getInstance(this).addToRequestQueue(stringRequest);
+```
+
+One more important aspect of Volley: **you don't have to specify for requests to be done asynchronously!** Volley handles all requests on a separate thread and deals with multi-threading for you, and returns all responses to the main thread.
+
+## Assignment
+Your homework is to create an application using Twitter's (RESTful API)[https://dev.twitter.com/rest/public/search]. Create a UI that allows the user to search for a given tweet by phrase, and displays 5 results. Allow the user to swipe right to dismiss these 5 results and bring up the next 5. If they swipe left, you should bring up the previous 5. If the user taps a tweet, open the tweet in the user's web browser or twitter client.
+
+Instead of using local storage for the app you've made in previous classes, switch to using S3.
```

Comment:
@MobileProto16/teaching Perhaps we can have them take a previous app and add a toggle button for storage in SQL or Amazon S3? And then for this homework they would have to add S3? Also @nmohamed perhaps we should ask them to integrate somehow with one of the built in APIs on their phones, either into the app I described or into an old one. (I know that we planned camera and accelerometer later, but honestly I feel like we could just ask them to figure it out now.)